report/renderer/details-renderer.js

```diff
@@ -394,29 +496,48 @@ export class DetailsRenderer {
       this._dom.createChildOf(theadTrElem, 'th', classes).append(labelEl);
     }
 
+    const entityItems = this._getEntityGroupItems(details);
     const tbodyElem = this._dom.createChildOf(tableElem, 'tbody');
-    let even = true;
-    for (const item of details.items) {
-      const rowsFragment = this._renderTableRowsFromItem(item, details.headings);
-
-      // The attribute item.entity could be a string (entity-classification), or
-      // a LinkValue for ThirdPartySummary audit.
-      let entityName;
-      if (typeof item.entity === 'object' && item.entity.type === 'link') {
-        entityName = item.entity.text;
-      } else if (typeof item.entity === 'string') {
-        entityName = item.entity;
+    if (entityItems.length) {
+      for (const entityItem of entityItems) {
+        const entityName = typeof entityItem.entity === 'string' ? entityItem.entity : undefined;
+        // Render the heading row
+        const entityGroupFragment = this._renderEntityGroupRow(entityItem, details.headings);
+        // Render all the items that match the heading row
+        for (const item of details.items.filter((item) => item.entity === entityName)) {
+          entityGroupFragment.append(this._renderTableRowsFromItem(item, details.headings));
+        }
+        const rowEls = this._dom.findAll('tr', entityGroupFragment);
+        const firstRowEl = rowEls[0];
+        firstRowEl.classList.add('lh-row--group');
+        if (entityName) {
+          rowEls.forEach(row => row.dataset.entity = entityName);
+          this._adornTableRowWithEntityChips(firstRowEl);
+        }
+        tbodyElem.append(entityGroupFragment);
```

Comment:
Good thought to use the existing zebra-striping on `.lh-row--group`, makes the table aggregation clearer. 👍 

It's unfortunate that we'll have a different UI approach for separating items based on if aggregated or not, though (when not aggregated: zebra. otherwise: nothing...). Items with long subitems next to each other really benefit from this striping, which is lost here. so I think it's worth exploring this more. What are some other options? think border-bottom? some sort of iconography before the entity name?

Reply:
Yeah we tried a few combinations of these with Peter/UX, and you can see the other attempts [here](https://docs.google.com/presentation/d/1iXm0xCe_Bd5lAipIGYdv5lWup1H6OpICa254-ZKQiCs/edit#slide=id.g18cc1102f3a_2_32), in the slides around it. This is cosmetic, so we could iterate if we have better options.

Reply:
thanks, I left a comment there.